lib/ttt.ex

```diff
@@ -5,55 +5,133 @@ defmodule TTT do
   end
 
   def new do
-    IO.puts "Let's play Tic Tac Toe!"
+    IO.puts "Let's play Tic Tac Toe!\n"
+    input = IO.gets "Please select a game mode: \n\n1. Player vs. Player\n2. Player vs. Computer\n3. Spectate a game\n\nYour choice is: "
+    String.trim(input)
+    |> game_mode()
+    |> start()
+  end
+
+  def game_mode(choice) do
+    cond do
+      choice == "1" -> player_player()
+      choice == "2" -> player_comp()
+      choice == "3" -> comp_comp()
+    end
+  end
+
+  def start({player1, player2} = _players) do
     board = Board.new
-    game = Game.new
+    game = Game.new(player1, player2)
     status = Game.status(game)
     Board.show(board)
-    play(board, game, status, :x)
+    play(board, game, status, player1)
+  end
+
+  def player_player do
+    p1name = set_player_name(1)
+    p2name = set_player_name(2)
+    player1 = Player.new(p1name, :x, :human)
+    player2 = Player.new(p2name, :o, :human)
+    {player1, player2}
+  end
+
+  def player_comp do
+    p1name = set_player_name(1)
+    p2name = set_player_name(2)
+    player1 = Player.new(p1name, :x, :human)
+    player2 = Player.new(p2name, :o, :computer)
+    {player1, player2}
+  end
+
+  def comp_comp do
+    p1name = set_player_name(1)
+    p2name = set_player_name(2)
+    player1 = Player.new(p1name, :x, :computer)
+    player2 = Player.new(p2name, :o, :computer)
+    {player1, player2}
   end
 
-  def play(board, game, status, turn) when status == :underway do
-    input = get_input(turn)
-    move = match_input(String.trim(input))
-    with {:ok, game} <- Game.play_turn(game, turn, move) do
+  def play(board, game, status, %Player{name: _name, token: _token, type: type} = turn)
+    when status == :underway do
+    cond do
+      type == :human ->
+        input = get_move(turn)
+        move = match_input(String.trim(input))
+        make_a_move(board, game, turn, move)
+      type == :computer ->
+        move = generate_move(game, 1)
+        make_a_move(board, game, turn, move)
+    end
+  end
+
+  def play(_board, _game, {_progress, {outcome, person}} = status, _turn) when status != :underway do
```

Comment:
The arguments to this function are getting pretty complicated. Maybe consider breaking it up into pieces? Or using a different data structure here?